design/PROCESSES.md

```diff
@@ -0,0 +1,23 @@
+# Document chapters
+The following chapters/information must be included in the design document (according to the unofficial Snelting tips document):
+
+* Introduction. Gives an overview of the document. Explains separation into packages.
+* Detailed class descriptions.
+* Sequence diagrams to illustrate common processes.
+* Changes relative to the requirements document.
+* Large class diagram that shows all classes.
+* When a design pattern is used it must be mentioned.
+
+# Tools
+Everyone must use the same tools for the design phase because they are usually not compatible with one another.
+We decided on using PlantUML and Doxygen for the design phase.
+
+## Workflow
+* Write Python class/method stubs with docstrings and type hints.
+* Manually create UML sequence diagrams.
+* Manually create UML class diagrams that show relations between classes for single packages.
```

Comment:
We can just cut out the non-relevant parts from the file generated from py2puml, so I don't think this is very difficult (or we can patch it to include only a single package)

Reply:
What I mean by relations are the connections that show how many objects of class A go together with objects of type B.
Something like: A 1 ---- 4 B
Since this depends on the actual implementation we can't automatically generate these from stubs, correct?

Reply:
I see. I think we can generate these if we parse the constructor, right? I will try to make a patch for that. py2puml is already parsing class variables.

Reply:
I think this is relatively complicated and also not urgent.
How about we talk about it on Monday?

Reply:
You are right, I just remembered we have aggregation, composition and many types of associations, and differentiating them is not that simple.